### DISCUSSION TITLE
How to handle the wrangler.toml configuration file?

### DISCUSSION THREAD

**Discussion by TinsFox** [cloudflare/workers-sdk#7115] [Q&A] [answered]
I am using Cloudflare Workers to build open source projects, but every time I have a question, what do I do with the wrangler.toml file?

1. there is information in there that needs to be filled in by the user. 2. i use services like kv r2.
2. I use services such as kv r2, so I need to fill in my own values for this to work.
3. the wrangler.toml file is required for deployment, if not it will cause an error if there is no main entry.
4. I want to use [CI](https://developers.cloudflare.com/workers/ci-cd/builds/) to deploy workers.

This is a question I've been having since I started using Workers, can anyone help me? Anyone can help me?

```wrangler.toml
name = "github-host"
main = "src/index.ts"
compatibility_date = "2024-10-28"
compatibility_flags = ["nodejs_compat"]

# [vars]
# API_KEY = ""

# [[kv_namespaces]]
# binding = ""
# id = ""


# [[r2_buckets]]
# binding = "MY_BUCKET"
# bucket_name = "my-bucket"

# [[d1_databases]]
# binding = "DB"
# database_name = "my-database"
# database_id = ""

# [ai]
# binding = "AI"

[observability]
enabled = true
head_sampling_rate = 1

[triggers]
crons = ["0 * * * *"]
```

**Comment by TinsFox** [comment]
Can I safely disclose this information because others can't use it? 🤔

**Comment by emily-shen** [comment, marked as answer]
Hey! It's fine to have a public wrangler.toml and include values like KV namespace ids, account ids, bucket names etc.

However, if you have something like an API keys however, you should use [secrets](https://developers.cloudflare.com/workers/configuration/secrets/), which are an environment variable that is *not* included your wrangler.toml file. I note that you do seem to have an API key set using regular environment variables - you should definitely not commit this as-is!

Hope that helps :)

**Reply by TinsFox** [reply]
Thank you very much for your reply!
My API_KEY is stored in `.dev.vars` and added to .gitignore, which does not cause leakage. Thanks for the reminder!
